source/opt/instruction.h

```diff
@@ -91,6 +91,46 @@ inline bool operator!=(const Operand& o1, const Operand& o2) {
   return !(o1 == o2);
 }
 
+// This structure is used to represent a DebugScope instruction from
+// the OpenCL.100.DebugInfo extened instruction set. Since this
+// structure is added to Instruction class as a member variable, we
+// want to reduce the size of structure. Hence, we keep only its Scope
+// and InlinedAt operands but not its result id. Note that losing its
+// result id will result in a different binary even when there is no
+// optimization. We skip the sanity check in Optimizer::Run(..) when
+// the module contains a DebugScope instruction.
+class DebugScope {
+ public:
+  DebugScope(uint32_t id0, uint32_t id1)
+      : lexical_scope_(id0), inlined_at_(id1) {}
+
+  inline bool operator!=(const DebugScope& d) const {
+    return lexical_scope_ != d.lexical_scope_ || inlined_at_ != d.inlined_at_;
+  }
+
+  // Accessor functions for |lexical_scope_|.
+  uint32_t GetLexicalScope() const { return lexical_scope_; }
+  void SetLexicalScope(uint32_t scope) { lexical_scope_ = scope; }
+
+  // Accessor functions for |inlined_at_|.
+  uint32_t GetInlinedAt() const { return inlined_at_; }
+  void SetInlinedAt(uint32_t at) { inlined_at_ = at; }
+
+  // Pushes the binary segments for this DebugScope instruction into
+  // the back of *|binary|.
+  void ToBinary(uint32_t type_id, uint32_t result_id, uint32_t ext_set,
+                std::vector<uint32_t>* binary) const;
+
+ private:
+  // The result id of the lexical scope in which this debug scope is
+  // contained. The value is |0| if there is no scope.
```

Comment:
Sorry, I should have said this earlier.  I'm seeing too many `0` around the code.  Should be have a named constant for them: `kNoDebugScope` and `kNoInlinedAt`?  It might make the code more readable.